cmd/juju/application/upgradecharm.go

```diff
@@ -271,6 +273,20 @@ func (c *upgradeCharmCommand) Run(ctx *cmd.Context) error {
 		return errors.Trace(err)
 	}
 
+	// If the URL of the running application specifies no series, then use
+	// the series of the running application. This is needed since the
+	// application was likely deployed from a multiple-series charm that
+	// does not specify a series in its URL but other means like
+	// command line flag.
+	if oldURL.Series == "" {
```

Comment:
We could also get the charm name from the applicationInfo, but it doesn't currently return the charm URL (which is a bit of a pain).